libraries/lib-ipc/IPCServer.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <memory>
+
+class IPCChannel;
+class IPCChannelStatusCallback;
+
+/**
+ * \brief Simple TCP socket based ipc server. When created
+ * server starts listen for incoming connection (see IPCClient). 
+ */
+class IPC_API IPCServer final
+{
+   class Impl;
+   std::unique_ptr<Impl> mImpl;
+public:
+   /**
+    * \brief May fail with exception or with call to
+    * IPCChannelStatusCallback::OnConnectionError.
+    * Callback should be guaranteed to be alive
+    * until either IPCChannelStatusCallback::OnDisconnect
+    * or IPCChannelStatusCallback::OnConnectionError is called.
```

Comment:
Good, this explains that an exception may be thrown.  The other pull request that uses IPCServer now needs try-catch blocks.

Not clear perhaps from this explanation is that a call to `OnConnectionError` happens on another thread that is created during a successful construction.